src/ws/session.c

```diff
@@ -533,6 +537,45 @@ pam_conv_func_dummy (int num_msg,
   return PAM_CONV_ERR;
 }
 
+static void
+create_s4u_ticket (const char *username)
+{
+  OM_uint32 major, minor;
+  gss_buffer_desc user_buf = { .length = strlen (username), .value = (char *) username };
```

Comment:
Can you move this down to where it's actually used?  I find it difficult to understand the flow of information when it's all at the top like this...  Also: could this be `const`?

Reply:
Moved down the three declarations.

I'm afraid it can't be const:
```
src/ws/session.c:553:36: error: passing argument 2 of ‘gss_import_name’ discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
  553 |   major = gss_import_name (&minor, &user_buf, GSS_KRB5_NT_PRINCIPAL_NAME, &impersonee);
      |                                    ^~~~~~~~~
In file included from src/ws/session.c:28:
/usr/include/gssapi/gssapi.h:565:5: note: expected ‘gss_buffer_t’ {aka ‘struct gss_buffer_desc_struct *’} but argument is of type ‘const gss_buffer_desc *’ {aka ‘const struct gss_buffer_desc_struct *’}
  565 |     gss_buffer_t,       /* input_name_buffer */
      |     ^~~~~~~~~~~~
```

Reply:
Ah, moving all three down [fails](https://github.com/cockpit-project/cockpit/pull/16168/checks?check_run_id=3299309971) -- takes a moment to realize why, but it makes sense. So I only moved down `user_buf`.